slatedb/src/manifest/mod.rs

```diff
@@ -212,7 +212,7 @@ pub(crate) trait ManifestCodec: Send + Sync {
 
 impl Manifest {
     pub(crate) fn has_wal_sst_reference(&self, wal_sst_id: u64) -> bool {
-        wal_sst_id > self.core.replay_after_wal_id && wal_sst_id < self.core.next_wal_sst_id
+        wal_sst_id > self.core.replay_after_wal_id && wal_sst_id < self.core.last_seen_wal_id
```

Comment:
Should this be `<=`? Some tests to verify this would be nice (if they don't already exist)